br/pkg/storage/locking.go

```diff
@@ -67,58 +175,145 @@ func readLockMeta(ctx context.Context, storage ExternalStorage, path string) (Lo
 	return meta, nil
 }
 
-func putLockMeta(ctx context.Context, storage ExternalStorage, path string, meta LockMeta) error {
-	file, err := json.Marshal(meta)
-	if err != nil {
-		return errors.Annotatef(err, "failed to marshal lock meta %s", path)
-	}
-	err = storage.WriteFile(ctx, path, file)
+type RemoteLock struct {
+	txnID   uuid.UUID
+	storage ExternalStorage
+	path    string
+}
+
+func tryFetchRemoteLock(ctx context.Context, storage ExternalStorage, path string) error {
+	meta, err := readLockMeta(ctx, storage, path)
 	if err != nil {
-		return errors.Annotatef(err, "failed to write lock meta at %s", path)
+		return err
 	}
-	return nil
+	return ErrLocked{Meta: meta}
 }
 
 // TryLockRemote tries to create a "lock file" at the external storage.
 // If success, we will create a file at the path provided. So others may not access the file then.
 // Will return a `ErrLocked` if there is another process already creates the lock file.
 // This isn't a strict lock like flock in linux: that means, the lock might be forced removed by
 // manually deleting the "lock file" in external storage.
-func TryLockRemote(ctx context.Context, storage ExternalStorage, path, hint string) (err error) {
-	defer func() {
-		log.Info("Trying lock remote file.", zap.String("path", path), zap.String("hint", hint), logutil.ShortError(err))
-	}()
-	exists, err := storage.FileExists(ctx, path)
-	if err != nil {
-		return errors.Annotatef(err, "failed to check lock file %s exists", path)
-	}
-	if exists {
-		meta, err := readLockMeta(ctx, storage, path)
-		if err != nil {
-			return err
-		}
-		return ErrLocked{Meta: meta}
+func TryLockRemote(ctx context.Context, storage ExternalStorage, path, hint string) (lock RemoteLock, err error) {
```

Comment:
why not use `TryLockRemoteWrite` instead of `TryLockRemote`?

Reply:
Just for compatibility. Write locks requires an extra suffix.